### ISSUE TITLE
Provide 'delay' option to specify delays between consecutive request

### ISSUE THREAD

**Issue by ppaulweber** [Orange-OpenSource/hurl#1832]
### Problem to solve

In order to test time sensitive (related) behavioral patterns of servers `hurl` would need to express a "delay" in order to wait between to consecutive requests.

### Proposal

Provide a `TIMEOUT` keyword to express on `ms` (or even `ns`) granularity a timeout about to delay consecutive requests as depicted in the following example:

```
POST {{LOCATION}}/endpoint
{
  "key": "value"
}

TIMEOUT 150

GET {{LOCATION}}/endpoint
```

### Additional context and resources

This use case is related to describe more expressive and compete test/benchmark plans using combinations of the features proposed in https://github.com/Orange-OpenSource/hurl/issues/88#issuecomment-1674275854

**Comment by fabricereix** [comment]
Hi @ppaulweber ,
we plan to add a `delay`option in our `[Options]` section to delay the execution of the request.
```
POST {{LOCATION}}/endpoint
{
  "key": "value"
}

GET {{LOCATION}}/endpoint
[Options]
delay: 150
```

**Comment by ppaulweber** [comment]
@fabricereix sounds great. So from the example you provided the semantics would be that the `GET` request is delayed by 150ms, right?

**Comment by fabricereix** [comment]
exactly

**Comment by jcamiel** [comment]
Fixed by #1832
